schemas/nl.basjes/yauaa_context/jsonschema/1-0-3

```diff
@@ -0,0 +1,231 @@
+        "deviceName": {
+            "description": "Example: Google Nexus 6",
+            "type": "string",
+            "maxLength": 100
+        },
+        "deviceBrand": {
+            "description": "Example: Google",
+            "type": "string",
+            "maxLength": 50
+        },
+        "deviceCpu": {
+            "type": "string",
+            "maxLength": 50
+        },
+        "deviceCpuBits": {
+            "type": "string",
+            "maxLength": 20
+        },
+        "deviceFirmwareVersion": {
+            "type": "string",
+            "maxLength": 100
+        },
+        "deviceVersion": {
+            "type": "string",
+            "maxLength": 100
+        },
+        "operatingSystemClass": {
+            "description": "See https://yauaa.basjes.nl/README-Output.html",
+            "enum": ["Desktop", "Mobile", "Cloud", "Embedded", "Game Console", "Hacker", "Anonymized", "Unknown"]
+        },
+        "operatingSystemName": {
+            "description": "Examples: Linux, Android.",
+            "type": "string",
+            "maxLength": 100
+        },
+        "operatingSystemVersion": {
+            "type": "string",
+            "maxLength": 50
+        },
+        "operatingSystemNameVersion": {
+            "type": "string",
+            "maxLength": 150
+        },
+        "operatingSystemVersionBuild": {
+            "type": "string",
+            "maxLength": 100
+        },
+        "layoutEngineClass": {
+            "description": "See https://yauaa.basjes.nl/README-Output.html",
+            "enum": ["Browser", "Mobile App", "Hacker", "Robot", "Unknown", "Special", "Cloud", "eReader"]
+        },
+        "layoutEngineName": {
+            "type": "string",
+            "maxLength": 100
+        },
+        "layoutEngineVersion": {
+            "type": "string",
+            "maxLength": 50
+        },
+        "layoutEngineVersionMajor": {
+            "type": "string",
+            "maxLength": 20
+        },
+        "layoutEngineNameVersion": {
+            "type": "string",
+            "maxLength": 150
+        },
+        "layoutEngineNameVersionMajor": {
+            "type": "string",
+            "maxLength": 120
+        },
+        "layoutEngineBuild": {
+            "type": "string",
+            "maxLength": 100
+        },
+        "agentClass": {
+            "description": "See https://yauaa.basjes.nl/README-Output.html",
+            "enum": ["Browser", "Browser Webview", "Mobile App", "Robot", "Robot Mobile", "Cloud Application", "Email Client", "Voice", "Special", "Testclient", "Hacker", "Unknown", "Desktop App", "eReader"]
+        },
+        "agentName": {
+            "description": "Example: Chrome.",
+            "type": "string",
+            "maxLength": 100
+        },
+        "agentVersion": {
+            "type": "string",
+            "maxLength": 100
+        },
+        "agentVersionMajor": {
+            "type": "string",
+            "maxLength": 100
```

Comment:
Is this not technically a breaking change for SchemaVer?  Not sure Redshift for example can handle changing the length of a field without a major version bump?

Reply:
> Is this not technically a breaking change for SchemaVer?

No, because all historic data is compatible with the new schema, and so [according to this blog post](https://snowplowanalytics.com/blog/2014/05/13/introducing-schemaver-for-semantic-versioning-of-schemas/) that makes it an `addition` (final number).

> Not sure Redshift for example can handle changing the length of a field without a major version bump?

That used to be true, but [this issue in schema-ddl](https://github.com/snowplow/schema-ddl/issues/83) addressed that problem.  Starting from RDB loader 1.2.x we can alter the column length.  Same too for Postgres Loader > 0.3.0.

Reply:
Oh, my mistake.... looks like Postgres Loader does not handle this yet, although Redshift loader does.  I will need to rethink this.

Reply:
Isn't that only true assuming as well that everyone using this enrichment and that updates enrich is _also_ using the latest RDB Loader which they might well not be.  If they are using an older loader version as well won't this break as it will find all the schema paths and try to push for an upgrade regardless?

Reply:
The important thing is that we can provide an upgrade path to open source users.  In this case, the path is to upgrade RDB loader to >1.2.0 and then upgrade enrich to 3.0.0 (released soon).  Alternatively, an open source user may choose to use an older version of RDB loader, and enrich 2.x.

Sooner or later though, we need to start taking advantage of the feature that our modern loaders can alter column max lengths.

Reply:
I have a vague recollection that we've previously increased the length of a string before, in a patch release, with the knowledge that the event would still fail to load in some warehouses but would at least now pass validation (and load into some more flexible like BigQuery). You'd also have to upgrade your enrich version (v3) to start using this new schema, and with that we could caveat that you also need to be on RDB/PG Loader x.x.x too.

Reply:
We could also make this 2-0-0 since it'd be introduced with Enrich v3... more of a challenging upgrade path for all users when we do that from a data modelling point of view though...

Reply:
Resolving based on this discussion: https://snowplow.slack.com/archives/GPFKE9CG1/p1637582919008300